pkg/controller/job/jobcontroller.go

```diff
@@ -419,7 +424,7 @@ func (jm *JobController) syncJob(key string) error {
 		job.Status.Failed = failed
 
 		if err := jm.updateHandler(&job); err != nil {
-			glog.Errorf("Failed to update job %v, requeuing.  Error: %v", job.Name, err)
+			utilruntime.HandleError(fmt.Errorf("Failed to update job %v, requeuing.  Error: %v", job.Name, err))
```

Comment:
@soltysh If you agree, I will remove these [lines](https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/job/jobcontroller.go#L327-L331) and simply use `key` we passed to [syncJob](https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/job/jobcontroller.go#L297) to replace `jobKey`. Then, I will simply use `return err` to replace these [lines](https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/job/jobcontroller.go#L421-L422)